irr/src/CIrrDeviceSDL.cpp

```diff
@@ -267,6 +267,32 @@ int CIrrDeviceSDL::findCharToPassToIrrlicht(uint32_t sdlKey, EKEY_CODE irrlichtK
 	}
 }
 
+u32 CIrrDeviceSDL::getScancodeFromKey(const KeyCode &key) const
+{
+	u32 keynum = 0;
+	if (key.index() == 0) {
+		auto keycode = std::get<EKEY_CODE>(key);
+		for (const auto &entry: KeyMap) {
+			if (entry.second == keycode) {
+				keynum = entry.first;
+				break;
+			}
+		}
+	} else {
+		keynum = std::get<wchar_t>(key);
+	}
+	return SDL_GetScancodeFromKey(keynum);
+}
+
+KeyCode CIrrDeviceSDL::getKeyFromScancode(const u32 scancode) const
+{
+	auto keycode = SDL_GetKeyFromScancode((SDL_Scancode)scancode);
```

Comment:
Why not make the parameter `SDL_Scancode` to begin with?

Reply:
Update now that I think about this again: another approach here would be to limit scancodes to SDL and put scancode-related logic behind `#ifdef _IRR_COMPILE_WITH_SDL_DEVICE_` guards. This would also avoid issues arising from the awkward non-SDL scancode workaround in the current form of the PR.